### PR TITLE
Add `test_each_hash`

### PR DESCRIPTION
**Pull request by gdritter-stripe**
### Motivation
User request: using `test_each` with a hash. As per offline conversation, we felt it would be better to have a hash-specific helper method for this rather than rely on Ruby's multiple-block-argument behavior.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by aisamanra** [comment]
…opened this from the wrong account so it assigned it to me, so I'm re-rolling.

**Comment by aisamanra** [comment]
Right now, this doesn't raise an error about
```
test_each([[1, 2], [3, 4]]) do |x, y|
  # ...
end
```
but it now accepts it. (It technically doesn't do any argument-checking in the rewriter pass, leaving that all for the declared static types.) I was leaning towards implementing that, but it also would be kind of a special case for this rewrite specifically (as AFAIK we don't have other places where we explicitly outlaw using multiple arguments to a block to avoid destructuring): would that be worth doing here?

File: test/testdata/rewriter/minitest_tables.rb [hunk 1]
```diff
@@ -61,4 +84,19 @@ def self.it(name, &blk); end
       T.reveal_type(v) # error: Revealed type: `T.any(String, Integer, T::Hash[T.untyped, T.untyped])`
     end
   end
+
+  test_each_hash({foo: 1, bar: 2, baz: 3}) do |k, v|
```
Comment (DarkDimius) [inline test/testdata/rewriter/minitest_tables.rb]: Can we add an DSL exp file?

**Comment by DarkDimius** [comment]
> I was leaning towards implementing that, but it also would be kind of a special case for this rewrite specifically (as AFAIK we don't have other places where we explicitly outlaw using multiple arguments to a block to avoid destructuring):

I'd be wine with that.

**Pull request merged**